packages/js/src/settings/components/formik-page-select-field.js

```diff
@@ -0,0 +1,166 @@
+ */
+const PageSelectOptionsContent = ( { children, className = "" } ) => (
+	<div className={ classNames( "yst-flex yst-items-center yst-justify-center yst-gap-2 yst-py-2 yst-px-3", className ) }>
+		{ children }
+	</div>
+);
+
+PageSelectOptionsContent.propTypes = {
+	children: PropTypes.node.isRequired,
+	className: PropTypes.string,
+};
+
+/**
+ * @param {Object} props The props object.
+ * @param {string} props.name The field name.
+ * @param {string} props.id The field id.
+ * @param {string} props.className The className.
+ * @param {string} props.description The description.
+ * @returns {JSX.Element} The page select component.
+ */
+const FormikPageSelectField = ( { name, id, className = "", ...props } ) => {
+	const siteBasicsPolicies = useSelectSettings( "selectPreference", [], "siteBasicsPolicies", {} );
+	const pages = useSelectSettings( "selectPagesWith", [ siteBasicsPolicies ], values( siteBasicsPolicies ) );
+	const { addManyPages, fetchPages } = useDispatchSettings();
+	const [ { value, ...field }, , { setTouched, setValue } ] = useField( { type: "select", name, id, ...props } );
+	const [ status, setStatus ] = useState( ASYNC_ACTION_STATUS.idle );
+	const [ initialPagesFound, setInitialPagesFound ] = useState( false );
+	const [ queriedPageIds, setQueriedPageIds ] = useState( [] );
+	const canCreatePages = useSelectSettings( "selectPreference", [], "canCreatePages", false );
+	const createPageUrl = useSelectSettings( "selectPreference", [], "createPageUrl", "" );
+
+	const selectedPage = useMemo( () => {
+		const pageObjects = values( pages );
+		return find( pageObjects, [ "id", value ] );
+	}, [ value, pages ] );
+
+
+	const debouncedFetchPages = useCallback( debounce( async search => {
+		try {
+			setStatus( ASYNC_ACTION_STATUS.loading );
+			// Cleanup previous running request.
+			if ( abortController ) {
+				abortController?.abort();
+			}
+			abortController = new AbortController();
+			// eslint-disable-next-line camelcase
+			const response = await fetchPages( { search, per_page: 20 } );
+
+			setQueriedPageIds( map( response.payload, "id" ) );
+			setStatus( ASYNC_ACTION_STATUS.success );
+		} catch ( error ) {
+			if ( error instanceof DOMException && error.name === "AbortError" ) {
+				// Expected abort errors can be ignored.
+				return;
+			}
+			setQueriedPageIds( [] );
+			setStatus( ASYNC_ACTION_STATUS.error );
+		}
+	}, 200 ), [ setQueriedPageIds, addManyPages, setStatus ] );
+
+
+	const handleChange = useCallback( newValue => {
+		setTouched( true, false );
+		setValue( newValue );
+	}, [ setValue ] );
+	const handleQueryChange = useCallback( event => debouncedFetchPages( event.target.value ), [ debouncedFetchPages ] );
+
+	useEffect( () => {
+		// Get initial options.
+		if ( ! initialPagesFound && ! isEmpty( pages ) ) {
+			setQueriedPageIds( map( pages, page => page.id ) );
+			setInitialPagesFound( true );
+		}
+	}, [ pages, initialPagesFound ] );
```

Comment:
Solved this in the PHP by appending 10 pages to the existing list there since this component is reused multiple times and I don't want more api calls.

Reply:
Regarding this solution, we don't `preparePage` the pages that come upon page load via the `append_newest_posts()` method in preferences.

This means for example that a Page with no title will appear as empty at first, but when the initial fetch request completes, it will automatically appear with its slug:
![image](https://github.com/Yoast/wordpress-seo/assets/12400734/d3c1fdea-bc60-403f-9c6a-35833bb2147d)
(^ before the initial fetch requests completes)

![image](https://github.com/Yoast/wordpress-seo/assets/12400734/366d737a-f799-4824-b03c-c7a16dbe7010)
(^ after the initial fetch requests completes)